tests/core/math/test_projection.h

```diff
@@ -0,0 +1,664 @@
+		Projection actual = Projection(
+				Vector4(-29.32f, 43.39f, -94.85f, -89.95f),
+				Vector4(-55.01f, 55.50f, -87.92f, -36.67f),
+				Vector4(48.63f, 19.80f, 75.48f, 44.12f),
+				Vector4(-2.16f, 70.96f, -59.63f, 23.87f))
+									.create_depth_correction(true);
+
+		for (int i = 0; i < 4; i++) {
+			for (int j = 0; j < 4; j++) {
+				CHECK(expected[i][j] == doctest::Approx(actual[i][j]));
+			}
+		}
+	}
+
+	SUBCASE("Without vertical flip") {
+		Projection expected(Vector4(1, 0, 0, 0), Vector4(0, 1, 0, 0), Vector4(0, 0, 0.5f, 0), Vector4(0, 0, 0.5, 1.0f));
+		Projection actual = Projection(
+				Vector4(-29.32f, 43.39f, -94.85f, -89.95f),
+				Vector4(-55.01f, 55.50f, -87.92f, -36.67f),
+				Vector4(48.63f, 19.80f, 75.48f, 44.12f),
+				Vector4(-2.16f, 70.96f, -59.63f, 23.87f))
+									.create_depth_correction(false);
+
+		for (int i = 0; i < 4; i++) {
+			for (int j = 0; j < 4; j++) {
+				CHECK(expected[i][j] == doctest::Approx(actual[i][j]));
+			}
+		}
+	}
+}
+
+TEST_CASE("[Projection] create_fit_aabb") {
+	AABB aabb(Vector3(1.0f, 1.0f, 1.0f), Vector3(2.0f, 2.0f, 2.0f));
+	Projection expected(
+			Vector4(1.0f, 0.0f, 0.0f, 0.0f),
+			Vector4(0.0f, 1.0f, 0.0f, 0.0f),
+			Vector4(0.0f, 0.0f, 1.0f, 0.0f),
+			Vector4(-2.0f, -2.0f, -2.0f, 1.0f));
+	Projection actual = Projection().create_fit_aabb(aabb);
+
+	for (int i = 0; i < 4; i++) {
+		for (int j = 0; j < 4; j++) {
+			CHECK(expected[i][j] == doctest::Approx(actual[i][j]));
+		}
+	}
+}
+
+TEST_CASE("[Projection] create_for_hmd") {
+	SUBCASE("Projection for eye 1") {
+		Projection expected(
+				Vector4(0.735632f, 0.0f, 0.0f, 0.0f),
+				Vector4(0.0f, 0.735632f, 0.0f, 0.0f),
+				Vector4(-0.114943f, 0.0f, -3.0f, -1.0f),
+				Vector4(0.0f, 0.0f, -4.0f, 0.0f));
+		Projection actual = Projection().create_for_hmd(1, 1.0f, 6.0f, 14.5f, 4.0f, 1.5f, 1.0f, 2.0f);
+
+		for (int i = 0; i < 4; i++) {
+			for (int j = 0; j < 4; j++) {
+				CHECK(expected[i][j] == doctest::Approx(actual[i][j]));
+			}
+		}
+	}
+
+	SUBCASE("Projection for eye 2") {
+		Projection expected(
+				Vector4(0.735632f, 0.0f, 0.0f, 0.0f),
+				Vector4(0.0f, 0.735632f, 0.0f, 0.0f),
+				Vector4(0.114943f, 0.0f, -3.0f, -1.0f),
+				Vector4(0.0f, 0.0f, -4.0f, 0.0f));
+		Projection actual = Projection().create_for_hmd(2, 1.0f, 6.0f, 14.5f, 4.0f, 1.5f, 1.0f, 2.0f);
+
+		for (int i = 0; i < 4; i++) {
+			for (int j = 0; j < 4; j++) {
+				CHECK(expected[i][j] == doctest::Approx(actual[i][j]));
+			}
+		}
+	}
+}
+
+TEST_CASE("[Projection] create_frustum") {
+	SUBCASE("Invalid parameter p_right <= p_left") {
+		ERR_PRINT_OFF;
+		Projection actual = Projection().create_frustum(5.098039f, 5.098039f, -2.666666f, 2.666666f, 1.0f, 2.0f);
+		ERR_PRINT_ON;
+
+		Projection expected(Vector4(1, 0, 0, 0),
+				Vector4(0, 1, 0, 0),
+				Vector4(0, 0, 1, 0),
+				Vector4(0, 0, 0, 1));
+				
+		for (int i = 0; i < 4; i++) {
+			for (int j = 0; j < 4; j++) {
+				CHECK(expected[i][j] == doctest::Approx(actual[i][j]));
+			}
+		}
```

Comment:
I'd suggest adding a method for testing against identity to reduce redundant code

Reply:
Method to test against identity added. 
I would be happy to add these functions to the `Projection` class. Do I have to open an issue to propose these additions?

Reply:
That'd best be handled by a separate PR

Also testing against identity is easier to do like `doctest::Approx(i==j ? 1.0 : 0.0)`

Reply:
Should I open an issue before opening a new PR with the changes related to the missing methods of `Projection`?

Reply:
I think they are minor enough to allow opening a PR directly, and open a proposal if prompted to do so